### PR TITLE
Proposed solution

### PR DESCRIPTION
**Pull request by selfbd**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -34,29 +35,24 @@ struct T
...
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
-        return nullptr;
+        /*  B. Self, 4/12/2020
+        Suggesting scenario of both values being equal is outside the scope of this exercise.
+        If this case was relevant however, might handle using epsilon check such as the following:
+        if (std::abs(a.value - b.value) < 0.001f)       // handle case of both values being equal
+        */
+        return (a.value < b.value) ? a : b;
```
Comment (matkatmusic) [inline main.cpp:45]: re-read instruction 1
- [ ] revise according to spec

File: main.cpp [hunk 2]
```diff
@@ -69,63 +65,42 @@ struct U
...
 };
 
 int main()
 {
     T person1(76, "Neil");
+    T& person1Ref = person1;
     T person2(70, "Geddy");
-    
+    T& person2Ref = person2;
+
```
Comment (matkatmusic) [inline main.cpp:88]: no need to create references in order to pass references around
- [ ] remove these reference variables

File: main.cpp [hunk 3]
```diff
@@ -69,63 +65,42 @@ struct U
...
+    auto& smaller = f.compare(person1Ref, person2Ref);
+    std::cout << "the smaller one is " << smaller.name << std::endl;
 
     std::cout << std::endl;
     U controlPoint1;
+    U& controlPoint1Ref = controlPoint1;
     float updatedValue = 5.f;
-    std::cout << "Controller::control controlPoint1's multiplied values:\n" << Controller::control(&controlPoint1, &updatedValue) << std::endl;
-    
+    float& updatedValueRef = updatedValue;
```
Comment (matkatmusic) [inline main.cpp:98]: - [ ] remove these reference variables

File: main.cpp (hunk 1 above)
Comment (selfbd) [reply main.cpp]: Returned MinHeight.compare()to using pointers instead of references

File: main.cpp (hunk 2 above)
Comment (selfbd) [reply main.cpp]: Removed unnecessary reference variables from main()

File: main.cpp (hunk 3 above)
Comment (selfbd) [reply main.cpp]: Removed unnecessary reference variables from main()

File: main.cpp [hunk 4]
```diff
@@ -35,14 +35,14 @@ struct T
...
-        If this case was relevant however, might handle using epsilon check such as the following:
-        if (std::abs(a.value - b.value) < 0.001f)       // handle case of both values being equal
-        */
-        return (a.value < b.value) ? a : b;
+        if (a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
+        return nullptr;
```
Comment (matkatmusic) [inline main.cpp]: - [ ] Change function parameters to be references
leave return type as is

Comment (selfbd) [reply main.cpp]: Changed parameters from pointers to references in MinHeight.compare()

**Pull request merged**